src/UbiHttp.cpp

```diff
@@ -157,79 +166,62 @@ double UbiHTTP::get(const char *device_label, const char *variable_label) {
   }
 
   _client_https_ubi.print(message);
-
-  while (_client_https_ubi.connected()) {
-    if (_client_https_ubi.available()) {
-      String line = _client_https_ubi.readStringUntil('\n');
-      if (line == "\r") {
-        break;
-      }
-    }
-  }
-
+  _client_https_ubi.flush();
+  
   free(message);
   free(path);
-
-  double value = _parseServerAnswer();
-
-  _client_https_ubi.flush();
-  _client_https_ubi.stop();
+  
+  double value = NAN;
+  
+  /* Reads the response from the server */
+  if (waitServerAnswer()) {
+    value = _parseServerAnswer();
+  } else {
+    if (_debug) {
+      Serial.println(F("Could not read server's response"));
+    }
+  }
+  
   return value;
 }
 
 double UbiHTTP::_parseServerAnswer() {
-  /**
-   * @param _charResponseLength 3 is the maximun amount of digits for the length
-   * to have
-   */
-  char *_charLength = (char *)malloc(sizeof(char) * 3);
-
-  _parsePartialServerAnswer(_charLength);
-  /**
-   * The server respond the length of the value in HEX so it has to be converted
-   * to DEC
-   * */
-  uint8_t length = UbiUtils::hexadecimalToDecimal(_charLength);
-  char *_charValue = (char *)malloc(sizeof(char) * length + 1);
-
-  _parsePartialServerAnswer(_charValue);
-
-  double value = strtof(_charValue, NULL);
-
-  free(_charLength);
-  free(_charValue);
+  /* Skip response headers */
+  if (_debug) {
+    Serial.println("Server response:");
+  }
+  while (_client_https_ubi.connected()) {
+    String line = _client_https_ubi.readStringUntil('\n');
```

Comment:
Probably the ESP32 has sufficient RAM to just spend it like this, the String object if like a PACMAN of RAM, That's why the previous implementation was doing.

The final goal of this is to unify the libraries at some point, probably some of them will have different implementations, but if we can develop the library thinking about the device with the less amount of RAM available this implementation may break due overflowing.

Reply:
I'd rather change this for something less RAM expensive.

Reply:
The library already use one instance of String class, I will keep that one.

Reply:
@crisap94 @deltalejo  to use a String buffer to read the server's response is a RAM waste, I think that you should not add the piece of code of line 194